src/TeleportRouter.sol

```diff
@@ -120,37 +128,38 @@ contract TeleportRouter {
      */
-    function requestMint(
-        TeleportGUID calldata teleportGUID,
-        uint256 maxFeePercentage,
-        uint256 operatorFee
-    ) external returns (uint256 postFeeAmount, uint256 totalFee) {
-        require(msg.sender == gateways[teleportGUID.sourceDomain], "TeleportRouter/sender-not-gateway");
+    function registerMint(TeleportGUID calldata teleportGUID) external {
+        // We trust the parent gateway with any sourceDomain as a compromised parent implies compromised child
+        // Otherwise we restrict passing messages only from the actual source domain
+        require(msg.sender == parent || msg.sender == gateways[teleportGUID.sourceDomain], "TeleportRouter/sender-not-gateway");
         address gateway = gateways[teleportGUID.targetDomain];
+        // Use fallback if no gateway is configured for the target domain
+        if (gateway == address(0)) gateway = parent;
         require(gateway != address(0), "TeleportRouter/unsupported-target-domain");
-        (postFeeAmount, totalFee) = GatewayLike(gateway).requestMint(teleportGUID, maxFeePercentage, operatorFee);
+        GatewayLike(gateway).registerMint(teleportGUID);
     }
 
     /**
      * @notice Call a GatewayLike contract to settle a batch of sourceDomain -> targetDomain DAI transfer. 
      * The sender must be a supported gateway
-     * @param targetDomain The domain receiving the batch of DAI (only L1 supported for now)
-     * @param batchedDaiToFlush The amount of DAI in the batch 
+     * @param sourceDomain The domain sending the batch of DAI
+     * @param targetDomain The domain receiving the batch of DAI
+     * @param amount The amount of DAI in the batch 
      */
-    function settle(bytes32 targetDomain, uint256 batchedDaiToFlush) external {
-        bytes32 sourceDomain = domains[msg.sender];
-        require(sourceDomain != bytes32(0), "TeleportRouter/sender-not-gateway");
+    function settle(bytes32 sourceDomain, bytes32 targetDomain, uint256 amount) external {
+        // We trust the parent gateway with any sourceDomain as a compromised parent implies compromised child
+        // Otherwise we restrict passing messages only from the actual source domain
+        require(msg.sender == parent || msg.sender == gateways[sourceDomain], "TeleportRouter/sender-not-gateway");
         address gateway = gateways[targetDomain];
+        // Use fallback if no gateway is configured for the target domain
+        if (gateway == address(0)) gateway = parent;
         require(gateway != address(0), "TeleportRouter/unsupported-target-domain");
-         // Forward the DAI to settle to the gateway contract
-        dai.transferFrom(msg.sender, gateway, batchedDaiToFlush);
-        GatewayLike(gateway).settle(sourceDomain, batchedDaiToFlush);
+        // Forward the DAI to settle to the gateway contract
+        dai.transferFrom(msg.sender, address(this), amount);
+        dai.approve(gateway, amount);
```

Comment:
I think it should be fine to do an infinite approval of the target gateway when it is `file`'d. It seems to me that a compromised target domain could not steal that Dai as the `guestOnly` methods of `DomainHost` only ever transfer Dai from the escrow.

Reply:
Ah yes good point.